tensorflow/core/nccl/nccl_manager.h

```diff
@@ -78,6 +79,17 @@ class NcclManager {
       DCHECK(event_mgr != nullptr);
       DCHECK(tensor_stream != nullptr);
       DCHECK(nccl_stream != nullptr);
+      if (input != nullptr) {
+        input_event = new se::Event(executor);
```

Comment:
@jeffdaily what if we use `unique_ptr`?

Reply:
Good call.  I will change this to `absl::make_unique` and remove the explicit `Participant` destructor.

Reply:
@jeffdaily CI test result looks fine. failures in `rocm-xla` should have no relationship to this PR. please revise the code and we can take it.